### DISCUSSION TITLE
Support server-side content delivery like Apache Guacamole

### DISCUSSION THREAD

**Discussion by Sesshoumaru-sama** [linuxserver/Heimdall#700] [Ideas]
This is probably more of a feature idea then an issue - please excuse if there is another way to do this.

Usecase:
Imagine you open a port in your router to access Heimdall from anywhere in the world as your central application dashboard to your private infrastructure and the various services you host (e.g. nextcloud). You log-in and see the dashboard of Heimdall. However, the URL or even IP address of the application you want to access is only valid in your local network, not the public network you are comming from. This means that when opened in another tab nothing will happen (if that URL is not publically exposed as well).

The Idea:
Open the selected application inside something like an Iframe, the actual request (http/s) is executed on the Heimdall backend which is part of the local net, then delivered to the client from the Heimdall server - acting as proxy into the local network. This way you can add applications to the dashboard which are only locally available, turning Heimdall in a true remote access gatway.

Current Situation:
Currently you could use things like Apache Guacamole to get into the local net, from there RDP/VNC into a machine in the local network and there use Heimdall to access local services / applications. Its a hassle. Integration into Heimdall could open a set of new possibilites for the software.

**Comment by orbikm** [comment]
I have my home services set up behind a nginx reverse proxy, which is exposed to the internet via port forwarding.

Here's nginx's manual on setting up a [reverse proxy](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/).
